File: test_supabase.ts
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
dotenv.config();

const supabase = createClient(process.env.VITE_SUPABASE_URL!, process.env.VITE_SUPABASE_ANON_KEY!);

async function check() {
  const { data, error } = await supabase.from('feedback').select('*').limit(1);
  if (error) {
    console.error("Error fetching feedback:", error);
  } else {
    console.log("Feedback row:", data);
  }

  // Also try inserting a test row to see what fails
  const { error: insertError } = await supabase.from('feedback').insert([{
    user_name: 'Test',
    user_email: 'test@upl',
    rating: 5,
    message: 'Test message',
    type: 'Test',
    created_at: new Date().toISOString()
  }]);
  
  if (insertError) {
    console.error("Insert error:", insertError.message);
  } else {
    console.log("Insert success!");
  }
}

check();

File: test_supabase.py
import os
import requests

url = "https://lobghkggwkygogzmfnlk.supabase.co/rest/v1/feedback?select=*&limit=1"
with open('.env') as f:
    env_vars = dict(line.strip().split('=', 1) for line in f if '=' in line)

key = env_vars.get('VITE_SUPABASE_ANON_KEY')

headers = {
    "apikey": key,
    "Authorization": f"Bearer {key}"
}

try:
    response = requests.get(url, headers=headers)
    print("GET Status:", response.status_code)
    print("GET Response:", response.text)

    # Try inserting
    post_url = "https://lobghkggwkygogzmfnlk.supabase.co/rest/v1/feedback"
    data = {"user_name": "Test", "user_email": "test@upl", "rating": 5, "message": "Test message", "type": "Test"}
    post_res = requests.post(post_url, headers={**headers, "Content-Type": "application/json", "Prefer": "return=representation"}, json=data)
    print("POST Status:", post_res.status_code)
    print("POST Response:", post_res.text)

except Exception as e:
    print("Error:", e)
